### PR TITLE
Change zoom constraint setting and fix canvas aspect ratio resizing

### PR DESCRIPTION
**Pull request by k3vnchen**
Added constants for max and min zoom, as well as far and near clipping. Removed instances of `self.w` and `self.h` in favor of `self.size` and `self.aspect_ratio`. Modified `setProjectionMatrix` and `DoSetViewport` to fix window sizing issues with distortion.

Will need to change where setProjectionMatrix is in the future, as it is being called in `OnDraw`, whereas it may be better if it is only called during resize/scroll wheel events.

### CODE REVIEW

File: components/canvas.py [hunk 1]
```diff
@@ -1,20 +1,26 @@
 from OpenGL.GL import *
 from OpenGL.GLU import *
 import wx
-from wx import glcanvas
+from wx.glcanvas import GLCanvas
+from wx.glcanvas import GLContext
```
Comment (parksong22) [inline components/canvas.py:5]: This is minor thing. To make these two lines into one line, you can import two classes through one 'from' statement by separating  them by comma.

File: components/canvas.py [hunk 2]
```diff
@@ -39,12 +45,9 @@ def DoSetViewport(self):
         width, height = size = self.size = self.GetClientSize()
         self.SetCurrent(self.context)
 
-        # set viewport dimensions to square (may change later)
-        min_size = min(width, height)
-        glViewport(0, 0, min_size, min_size)
+        glViewport(0, 0, width, height)
+        self.aspect_ratio = width / height;
```
Comment (parksong22) [inline components/canvas.py:49]: Can you make sure if min_size variable is in use? If not, please remove the variable and any commented out code using min_size.

Comment (k3vnchen) [reply components/canvas.py:49]: I've already removed the line containing min_size (shown in red), and there aren't any more instances of it

**Comment by k3vnchen** [comment]
Hold on - don't merge just yet. Fixing something real quick

**Pull request merged**